File: connectors/integration-tests/src/test/scala/com/taskadapter/integrationtests/Adapter.scala
package com.taskadapter.integrationtests

import com.taskadapter.connector.common.ProgressMonitorUtils
import com.taskadapter.connector.definition.SaveResult
import com.taskadapter.connector.{FieldRow, NewConnector}
import com.taskadapter.core.PreviouslyCreatedTasksResolver

class Adapter(connector1: NewConnector, connector2: NewConnector) {

  def adapt(rows: List[FieldRow]): SaveResult = {
    val tasks = connector1.loadData()
    val result = connector2.saveData(PreviouslyCreatedTasksResolver.empty, tasks, ProgressMonitorUtils.DUMMY_MONITOR, rows)
    result
  }

}

File: connectors/integration-tests/src/test/java/com/taskadapter/core/Adapter.java
package com.taskadapter.core;

import com.taskadapter.connector.FieldRow;
import com.taskadapter.connector.NewConnector;
import com.taskadapter.connector.common.ProgressMonitorUtils;
import com.taskadapter.connector.definition.SaveResult;
import com.taskadapter.model.GTask;

import java.util.List;

class Adapter {
    final NewConnector connector1;
    final NewConnector connector2;

    public Adapter(NewConnector connector1, NewConnector connector2) {
        this.connector1 = connector1;
        this.connector2 = connector2;
    }

    SaveResult adapt(List<FieldRow<?>> rows) {
        List<GTask> tasks = connector1.loadData();
        return connector2.saveData(PreviouslyCreatedTasksResolver.empty(), tasks, ProgressMonitorUtils.DUMMY_MONITOR, rows);
    }
}

